<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mirrorDlg</class>
 <widget class="QDialog" name="mirrorDlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>757</width>
    <height>477</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Mirror configuration</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="7" column="0">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_14">
       <item>
        <widget class="QLabel" name="lblAutoInvert">
         <property name="text">
          <string>Autoinvert: not set</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnChangeAutoInvert">
         <property name="text">
          <string>Change...</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item row="0" column="0" colspan="4">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Name</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="name"/>
     </item>
    </layout>
   </item>
   <item row="3" column="1" colspan="3">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Fringe Spacing in waves:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="fringeSpacingEdit">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Set to 1 if using a Bath interferometer or similar.  Set to .5 if testing in autocollimation with same.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="5" column="1">
    <widget class="QPushButton" name="ReadBtn">
     <property name="text">
      <string>Read Existing File</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1" colspan="3">
    <widget class="QCheckBox" name="flipH">
     <property name="text">
      <string> Flip Interferogram left to right on loading </string>
     </property>
    </widget>
   </item>
   <item row="6" column="1" colspan="3">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Elliptical outline</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QCheckBox" name="ellipseShape">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Elliptical FLat or Nulled Optics</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label_9">
          <property name="acceptDrops">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string/>
          </property>
          <property name="text">
           <string>Vertical Axis</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="minorAxisEdit">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The vertical diameter of the Ellipse.&lt;/p&gt;&lt;p&gt; The other diameter which is always the horizontal diameter is specified by the Mirror diameter field. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QLabel" name="label_7">
     <property name="text">
      <string>Interferogram
Wave Length nm</string>
     </property>
    </widget>
   </item>
   <item row="2" column="2" colspan="2">
    <widget class="QLineEdit" name="z8">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Computed from mirror parameters and used when testing a non sphere at Radius of Curvature.&lt;/p&gt;&lt;p&gt;Uses Miirror Diameter, ROC Interferogram wave length, and desired conic constant.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </item>
   <item row="1" column="3">
    <widget class="QLineEdit" name="lambda"/>
   </item>
   <item row="7" column="1" colspan="3">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string/>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QCheckBox" name="useAnnulus">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;If the interferogram has a hole in the middle it is an interferogram of a mirror with a hole in the middle then annulus should be set and the diameter of the hole should be set either as a percentage of the mirror diameter or by the actual diameter of the hole.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Annulus</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="annularpercentLb">
          <property name="text">
           <string>Percent</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="annulusPercent">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Percentage of the hole compared to the mirror diameter.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_12">
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="annularDiamLb">
          <property name="text">
           <string>Diameter</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="annularDiameter">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Hole diameter.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="maximum">
           <double>10000.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_13">
        <item>
         <widget class="QPushButton" name="annulusHelp">
          <property name="text">
           <string>Help</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="8" column="0" colspan="4">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Help|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="1" column="0" rowspan="6">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Mirror </string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QCheckBox" name="unitsCB">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;If set units are displayed in mm.   If not, then units are in inches.  But only for this mirror configuration dialog.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Units in mm</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Diameter:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="diameter">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Diameter of clear aperture of the mirror or Horizontal axis of ellipse.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QLabel" name="rocLab">
          <property name="text">
           <string>Roc:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="roc">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Positive value of Radius of curvature of mirror or 0 if testing a flat. Will be ignored if Artificial Null is checked.&lt;/p&gt;&lt;p&gt;The Fnumber will be computed from this value and the diameter.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QLabel" name="fnumberLab">
          <property name="text">
           <string>FNumber:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="FNumber">
          <property name="toolTip">
           <string>Changing this value will compute the Roc value.</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Obstruction:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="obs">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Usually the size of the diagonal shadow on the mirror.  The surface error in this area will be ignored by the metrics.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Conic Constant:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="cc">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Desired conic Constant.  The actual value will be computed by the analysis.&lt;/p&gt;&lt;p&gt;This value is used in computing the Artiicial Null value.&lt;/p&gt;&lt;p&gt;Set to 0 if testing a flat or a sphere.&lt;/p&gt;&lt;p&gt;Set to -1 if testing a paraboloid.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QCheckBox" name="ReducApp">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enable masking the clear apature at the outside edge of the mirror.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Edge Mask</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="reduceValue">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Width of the edge to mask. New SA artificial null will be computed based on the clear aperature of the mirror with this this mask value.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>1000.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <widget class="QLabel" name="clearApLabel">
          <property name="text">
           <string>Unmasked Aperature:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="ClearAp">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QCheckBox" name="nullCB">
     <property name="text">
      <string>Artificial Null</string>
     </property>
    </widget>
   </item>
   <item row="5" column="3">
    <widget class="QPushButton" name="saveBtn">
     <property name="text">
      <string>Save in a File</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>mirrorDlg</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>mirrorDlg</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
